pymc3/plots/energyplot.py

```diff
@@ -0,0 +1,61 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from .utils import fast_kde
+
+def energyplot(trace, kind='kde', figsize=None, ax=None, legend=True, lw=0, alpha=0.5, **kwargs):
+    """Plot energy transition distribution and marginal energy distribution in order
+    to diagnose poor exploration by HMC algorithms.
+
+    Parameters
+    ----------
+
+    trace : result of MCMC run
+    kind : str
+        Type of plot to display (kde or histogram)
+    figsize : figure size tuple
+        If None, size is (8 x 6)
+    ax : axes
+        Matplotlib axes.
+    legend : bool
+        Flag for plotting legend (defaults to True)
+    lw : int
+        Line width
+    alpha : float
+        Alpha value for plot line. Defaults to 0.35.
+
+    Returns
+    -------
+
+    ax : matplotlib axes
+    """
+    
+    series_dict = {'Energy': trace['energy'] - trace['energy'].mean(),
```

Comment:
This will raise a rather uninformative `KeyError` if the step method doesn't have an energy statistic. Maybe we should catch and reraise or check first with `'energy' in trace.stat_names` to get a better error message.